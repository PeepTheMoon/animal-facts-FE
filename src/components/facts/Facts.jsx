import React from 'react';
import { useSelector } from 'react-redux';
import { getFacts } from '../selectors/factsSelectors';
import Fact from './Fact';

const Facts = () => {
  const facts = useSelector(getFacts);

  const factElements = facts.map(fact => (
    <li key={fact.id}>
      <Fact {...fact} />
    </li>
  ));

  return (
    <ul>
      {factElements}
    </ul>
  );
};

export default Facts;
