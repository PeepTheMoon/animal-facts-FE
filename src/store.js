import { createStore } from 'redux';
import reducer, { initialState } from './components/App/reducers/AppReducer';

const store = createStore(
  reducer, 
  initialState,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
