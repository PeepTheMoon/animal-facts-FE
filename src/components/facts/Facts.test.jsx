import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import store from '../../store';
import Facts from './Facts';

describe('Facts component', () => {
  it('renders Facts', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Facts />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
