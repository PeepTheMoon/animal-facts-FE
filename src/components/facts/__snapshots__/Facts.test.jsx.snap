// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facts component renders Facts 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Facts>
    <ul />
  </Facts>
</Provider>
`;
